### ISSUE TITLE
Code for WSJ0-REVERB dataset reproduction

### ISSUE THREAD

**Issue by stefan-baumann** [sp-uhh/sgmse#7]
Hi, thank you very much for the great work with the models and making this repository available. I am trying to reproduce some of the results from your paper and was wondering whether you could also make the code for creating the WSJ0-REVERB dataset as in your paper available. The paper already gives quite a bit of information but a release of the code to reproduce the exact dataset you used to train your model would be really appreciated as comparisons are difficult otherwise. Especially the test set would be good to have accessible for fair comparisons.

I also have some other questions about the models and this repo - I'll create some separate issues for them to not clutter up this one, I hope that's okay :)

All the best,
Stefan

**Comment by julius-richter** [comment]
Hi Stefan, thanks for your interest in our work!

We have added the code for the WSJ0-REVERB dataset creation in our latest commit. You'll find the script here: https://github.com/sp-uhh/sgmse/blob/main/preprocessing/create_wsj0_reverb.py

To run the script you need to set the arguments:
-  `--wsj0_dir` for the directory of the WSJ0 dataset which should contain the subdirectories "si_dt_05", "si_tr_s" and "si_et_05". 
- `--target_dir` for the path to the target directory for saving WSJ0-REVERB.

**Comment by stefan-baumann** [comment]
Hi Julius, thank you very much for the incredibly quick response and for sharing the code! I'll give it a try.

**Comment by stefan-baumann** [comment]
Hey, I've been trying to reproduce a training on the WSJ0-REVERB dataset created from the normal WSJ0 dataset and using the exact code you provided and am getting some weird behaviour: I see both some comb filtering-like artifacts in the target data and something that looks like a very short time delay between reverberant & anechoic in some samples. I can somewhat understand why there would be a time delay although it would make more sense to me if they were fully aligned, but I don't understand how you're getting the really clean samples presented on the project page with a model that's trained on these randomly comb-filtered targets.

Here's a grid of 16 anechoic samples and their reverberant counterparts from the dataset as generated using your script as a reference:
![wsj0_reverb_anechoic](https://user-images.githubusercontent.com/4400519/192792160-84a37dae-d381-4a87-94b8-c01f71f98168.png)
![wsj0_reverb_reverb](https://user-images.githubusercontent.com/4400519/192792164-9fb45ff9-3f57-4801-ba4f-5082ab3db8d3.png)

**Comment by jmlemercier** [comment]
Hi Stefan,
Thank you for spotting this, this is indeed a simple error coupled to a bug in the PyRoomAcoustics package.

We simplified the dataset generation code to post it here, and did a mistake in the default method used to simulate the reverberation. The code you used activates ray tracing in addition to the image source model. The problem apparently is that if you set a very high absorption coefficient for generating the "dry" room (corresponding to the so-called anechoic targets), ray tracing messes up the simulation, generating the comb filter artifacts displayed here.

We uploaded a new version of the code where only the image source model is used, deactivating the ray tracing method. The artifacts should disappear, and the results should be then consistent to the data we used for the approach presented in the paper.

Don't hesitate to reopen the issue if this does not solve the problem, for now I'll ask Julius to close this issue.

Best,
Jean-Marie
